nexus/db-queries/src/transaction_retry.rs

```diff
@@ -0,0 +1,141 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Helper types for performing automatic transaction retries
+
+use chrono::Utc;
+use oximeter::{types::Sample, Metric, MetricsError, Target};
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+
+// Identifies "which" transaction is retrying
+#[derive(Debug, Clone, Target)]
+struct Transaction {
+    name: String,
+}
+
+// Identifies that a retry has occurred, and track how long
+// the transaction took (either since starting, or since the last
+// retry failure was recorded).
+#[derive(Debug, Clone, Metric)]
+struct RetryData {
```

Comment:
Is it worth including some data that would let us know which attempt this latency corresponds to?

Reply:
Added!